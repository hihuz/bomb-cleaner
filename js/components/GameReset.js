import React from 'react';

const GameReset = props => (
  <div className="game-reset">
    <button className="reset-button" onClick={props.reset}>reset!</button>
  </div>
);

export default GameReset;
