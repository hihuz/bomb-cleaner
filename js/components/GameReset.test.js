import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import GameReset from './GameReset';

test('Snapshot', () => {
  const component = shallow(<GameReset />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Should call the reset prop when the reset-button is clicked', () => {
  const clickFn = jest.fn();
  const component = shallow(<GameReset reset={clickFn} />);
  component.find('button').simulate('click');
  expect(clickFn).toHaveBeenCalledTimes(1);
});
