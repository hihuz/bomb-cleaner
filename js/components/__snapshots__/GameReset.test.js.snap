exports[`test Snapshot 1`] = `
<div
  className="game-infos-item">
  <button
    className="reset-button">
    <i
      className="icon-refresh" />
  </button>
</div>
`;
